### PR TITLE
Fix oc secrets new-dockercfg to create secret identical like oc create secret

### PR DESCRIPTION
**Pull request by soltysh**
It looks like I missed one more create secret thing. Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1531511

/assign @juanvallejo

### CODE REVIEW

**Comment by mfojtik** [comment]
/lgtm

@juanvallejo FYI

**Comment by wanghaoran1988** [comment]
@soltysh IIRC this is for the old format docker config, does this mean we will only support the new format?

**Comment by juanvallejo** [comment]
@wanghaoran1988 
> IIRC this is for the old format docker config, does this mean we will only support the new format?

Right, I believe upstream now behaves the same way. The issue with the original move to do this was that we were incorrectly setting to secret type to still say it contained data in the old `dockercfg` format, and the key we were using was also for the old type, but we were actually storing secret data in the new dockercfgjson format.

**Comment by soltysh** [comment]
Updated the test-cmd failure, which were caused by these changes.

File: test/cmd/secrets.sh [hunk 1]
```diff
@@ -16,40 +16,40 @@ os::cmd::expect_failure_and_text 'oc secrets new foo --type=blah makefile=Makefi
 os::cmd::expect_success 'oc secrets new foo --type=blah makefile=Makefile --confirm'
 os::cmd::expect_success_and_text 'oc get secrets/foo -o jsonpath={.type}' 'blah'
 
-os::cmd::expect_success 'oc secrets new-dockercfg dockercfg --docker-username=sample-user --docker-password=sample-password --docker-email=fake@example.org'
+os::cmd::expect_success 'oc secrets new-dockercfg dockerconfigjson --docker-username=sample-user --docker-password=sample-password --docker-email=fake@example.org'
 # can't use a go template here because the output needs to be base64 decoded.  base64 isn't installed by default in all distros
-os::cmd::expect_success "oc get secrets/dockercfg -o jsonpath='{ .data.\.dockercfg }' | base64 -d > ${HOME}/dockerconfig"
-os::cmd::expect_success 'oc secrets new from-file .dockercfg=${HOME}/dockerconfig'
+os::cmd::expect_success "oc get secrets/dockerconfigjson -o jsonpath='{ .data.\.dockerconfigjson }' | base64 -d > ${HOME}/dockerconfigjson"
+os::cmd::expect_success 'oc secrets new from-file .dockerconfigjson=${HOME}/dockerconfigjson'
```
Comment (juanvallejo) [inline test/cmd/secrets.sh]: The test is okay, and although it did save to $HOME before this patch, I believe `${BASETMPDIR}` would be a more appropriate location to store `dockerconfigjson`

**Comment by juanvallejo** [comment]
one minor comment, but other than that
/lgtm

**Pull request merged**